kubeflow/trainer/rhai/transformers.py

```
@@ -22,15 +63,451 @@ class TransformersTrainer:
+    import json
+    import threading
+    import time
+    from typing import Any, Optional
+
+    from transformers import TrainerCallback, trainer as trainer_module
+
+    @dataclass
+    class ProgressionMetricsState:
+        """Progression metrics state (camelCase for Kubernetes API compatibility)."""
+
+        progressPercentage: Optional[int] = None  # noqa: N815
```

Ruff (0.14.5)

Unused `noqa` directive (non-enabled: `N815`)

Remove unused `noqa` directive

(RUF100)